transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueue.java

```diff
@@ -46,8 +49,29 @@
         if (cause != null) {
             UNAVAILABILITY_CAUSE = cause;
         } else {
-            UNAVAILABILITY_CAUSE = PlatformDependent.hasUnsafe() ? null :
+            cause = PlatformDependent.hasUnsafe() ? null :
                     new IllegalStateException("sun.misc.Unsafe not available");
+            if (cause == null) {
+                if (SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).contains("mac")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)